### PR TITLE
MONGOID-3551 Fix adding documents to nested relation using accepts_nested_attributes_for.

### PR DESCRIPTION
**Pull request by hck**

### CODE REVIEW

**Comment by estolfo** [comment]
Hi @hck
Thanks for this pull request. Do any of these [open tickets in our Mongoid Jira project](https://jira.mongodb.org/browse/MONGOID-4222?jql=text%20~%20%22accepts_nested_attributes_for%22%20and%20status%20%3D%20Open) look like they're related to the issue this PR resolves? If so, let me know so I can link this PR to it for better tracking. Otherwise, I'll create a new ticket for this (or you can, if you'd like)

Thanks!

**Comment by hck** [comment]
Hi, @estolfo
The fix is related to the issue in Jira, which was marked as "Cannot reproduce": https://jira.mongodb.org/browse/MONGOID-3551

I was able to reproduce it on Rails 4.2 and added necessary tests, which fail without a fix.

**Comment by estolfo** [comment]
Ok, thanks for that additional info. And you were using which version of Mongoid with Rails 4.2?

**Comment by hck** [comment]
@estolfo Mongoid 5.0.2 (also tried 5.0.1 and master branch from Github, the issue was reproducible in these versions also).

**Comment by estolfo** [comment]
Great, thanks! I'll try to reproduce it and look at the PR changes so they can be merged.

**Comment by hck** [comment]
@estolfo Thanks.

File: lib/mongoid/relations/builders/nested_attributes/many.rb [hunk 1]
```diff
@@ -167,6 +160,38 @@ def update_document(doc, attrs)
...
+          # @api private
+          #
+          # @example Update nested relation.
+          #   builder.update_nested_relation(parent, id, attrs)
+          #
+          # @param [ Document ] parent The parent document.
+          # @param [ String, BSON::ObjectId ] id of the related document.
+          # @param [ Hash ] attrs The single document attributes to process.
+          #
+          # @since 5.0.2
```
Comment (estolfo) [inline lib/mongoid/relations/builders/nested_attributes/many.rb]: The next version is 6.0 so you can write @since 6.0.0 here.

Comment (hck) [inline lib/mongoid/relations/builders/nested_attributes/many.rb]: @estolfo :+1:

**Pull request merged**